packages/c/sshnpd/src/daemon.c

```diff
@@ -0,0 +1,250 @@
+#include "sshnpd/device_info.h"
+#include "sshnpd/handle_npt_request.h"
+#include "sshnpd/handle_ping.h"
+#include "sshnpd/handle_ssh_request.h"
+#include "sshnpd/handle_sshpublickey.h"
+#include "sshnpd/permitopen.h"
+#include "sshnpd/sshnpd.h"
+#include <atchops/aes.h>
+#include <atchops/iv.h>
+#include <atchops/rsa.h>
+#include <atchops/rsa_key.h>
+#include <atchops/sha.h>
+#include <atclient/atclient.h>
+#include <atclient/atclient_utils.h>
+#include <atclient/atkey.h>
+#include <atclient/atkeys.h>
+#include <atclient/atkeys_file.h>
+#include <atclient/connection.h>
+#include <atclient/connection_hooks.h>
+#include <atclient/json.h>
+#include <atclient/monitor.h>
+#include <atclient/notify.h>
+#include <atclient/string_utils.h>
+#include <atlogger/atlogger.h>
+#include <libgen.h>
+#include <mbedtls/psa_util.h>
+#include <sshnpd/daemon.h>
+#include <sshnpd/file_utils.h>
+#include <sshnpd/run_srv_process.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#define LOGGER_TAG "sshnpd - loop"
+
+atclient worker;
+atclient monitor_ctx;
+char *ping_response;
+char *atserver_host;
+uint16_t atserver_port;
+atclient_atkeys atkeys;
+sshnpd_params params;
+char *regex;
+FILE *authkeys_file;
+char *authkeys_filename;
+char *home_dir;
+atchops_rsa_key_private_key signingkey;
+bool is_child_process;
+
+volatile sig_atomic_t should_run;
+
+// device info
+size_t device_info_pos;
+time_t *device_info_last_sent;
+uint8_t device_info_attempts;
```

Comment:
extern global state variables defined here. If they are defined directly in the header then certain compilers will see it as a duplicated definition.